src/execution_plan/ops/op_argument.c

```
@@ -7,63 +7,130 @@
-		OpBase_DeleteRecord(&arg->r);
+	OpArgument *arg = (OpArgument *)opBase;
+
+	// restore original record if 'r' was emitted
+	// and '_r' is present
+	if(arg->r == NULL && arg->_r != NULL) {
+		// restore original record
+		arg->r = arg->_r;
+
+		// create a copy for later restoration
+		arg->_r = OpBase_CreateRecord(opBase);
+		Record_Clone(arg->r, arg->_r);
```

Cppcheck (2.10-2)

[error] Memory leak

(memleak)